discovery_engine/test/ffi/types/history_test.dart

```diff
@@ -33,4 +33,25 @@ void main() {
     boxed.free();
     expect(res, equals(document));
   });
+
+  test('reading written Vec<HistoricDocument> works', () {
+    final docs = [
+      HistoricDocument(
+        id: DocumentId(),
+        url: Uri.parse('https://foo.example/'),
+        snippet: 'foo',
+        title: 'bar',
+      ),
+      HistoricDocument(
+        id: DocumentId(),
+        url: Uri.parse('https://foo.example/'),
+        snippet: 'dodo',
+        title: 'bird',
+      )
+    ];
+
+    final boxed = docs.allocNative();
+    final res = HistoricDocumentSliceFfi.consumeNative(boxed);
+    expect(res, equals(docs));
```

Comment:
the resource needs to be dropped
```suggestion
    boxed.free();
    expect(res, equals(docs));
```

Reply:
ah my bad, i missed that, nevermind.